WordPress/Classes/ViewRelated/Comments/CommentsViewController.m

```diff
@@ -158,114 +160,218 @@ - (NSString *)tableView:(UITableView *)tableView titleForHeaderInSection:(NSInte
+            [self.syncHelper syncMoreContent];
+        }
+    }
 }
 
-- (void)tableView:(UITableView *)tableView didCheckRowAtIndexPath:(NSIndexPath *)indexPath
+- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
 {
+    // Failsafe: Make sure that the Comment (still) exists
+    NSArray *sections = self.tableViewHandler.resultsController.sections;
+    if (indexPath.section >= sections.count) {
+        [tableView deselectSelectedRowWithAnimation:YES];
+        return;
+    }
+    
+    id<NSFetchedResultsSectionInfo> sectionInfo = sections[indexPath.section];
+    if (indexPath.row >= sectionInfo.numberOfObjects) {
+        [tableView deselectSelectedRowWithAnimation:YES];
+        return;
+    }
+    
+    // At last, push the details
+    Comment *comment            = [self.tableViewHandler.resultsController objectAtIndexPath:indexPath];
+    CommentViewController *vc   = [CommentViewController new];
+    vc.comment                  = comment;
+        
+    [self.navigationController pushViewController:vc animated:YES];
 }
 
-#pragma mark - Subclass methods
 
-- (NSString *)entityName
+#pragma mark - WPTableViewHandlerDelegate Methods
+
+- (NSManagedObjectContext *)managedObjectContext
 {
-    return @"Comment";
+    return [[ContextManager sharedInstance] mainContext];
 }
 
 - (NSFetchRequest *)fetchRequest
 {
-    NSFetchRequest *fetchRequest = [NSFetchRequest fetchRequestWithEntityName:[self entityName]];
-    fetchRequest.predicate = [NSPredicate predicateWithFormat:@"(blog == %@ AND status != %@)", self.blog, @"spam"];
-    NSSortDescriptor *sortDescriptorStatus = [NSSortDescriptor sortDescriptorWithKey:@"status" ascending:NO];
-    NSSortDescriptor *sortDescriptorDate = [NSSortDescriptor sortDescriptorWithKey:@"dateCreated" ascending:NO];
-    fetchRequest.sortDescriptors = @[sortDescriptorStatus, sortDescriptorDate];
-    fetchRequest.fetchBatchSize = 10;
+    NSFetchRequest *fetchRequest            = [NSFetchRequest fetchRequestWithEntityName:[self entityName]];
+    fetchRequest.predicate                  = [NSPredicate predicateWithFormat:@"(blog == %@ AND status != %@)", self.blog, CommentStatusSpam];
+    
+    NSSortDescriptor *sortDescriptorStatus  = [NSSortDescriptor sortDescriptorWithKey:@"status" ascending:NO];
+    NSSortDescriptor *sortDescriptorDate    = [NSSortDescriptor sortDescriptorWithKey:@"dateCreated" ascending:NO];
+    fetchRequest.sortDescriptors            = @[sortDescriptorStatus, sortDescriptorDate];
+    fetchRequest.fetchBatchSize             = CommentsFetchBatchSize;
+    
     return fetchRequest;
 }
 
+- (void)configureCell:(CommentsTableViewCell *)cell atIndexPath:(NSIndexPath *)indexPath
+{
+    Comment *comment = [self.tableViewHandler.resultsController objectAtIndexPath:indexPath];
+    cell.contentProvider = comment;
+}
+
 - (NSString *)sectionNameKeyPath
 {
     return @"status";
 }
 
-- (Class)cellClass
+- (NSString *)entityName
```

Comment:
This guy existed so WPTableViewController subclasses could specify what sort of content needed to be fetched.  If we wanted to, we could kill it to simplify things a bit.